src/sentry/static/sentry/app/types/iconProps.ts

```diff
@@ -0,0 +1,8 @@
+export interface IconProps extends React.SVGAttributes<SVGElement> {
+  color?: string;
+  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
+  direction?: 'up' | 'right' | 'down' | 'left';
+  solid?: boolean;
+  circle?: boolean;
```

Comment:
Not all of them always take `solid` or `circle` though, same with `type`.

Can we maybe remove these from the generic `IconProps` and just add them in as needed per icon component?

Reply:
I'm hoping anyone that adds more/new icons in the future will be able to consistently use solid and circle props again - I definitely think they'll be common enough. 

However, I will definitely remove an icon prop that I forgot about that's specifically for IconGraph  (below line 6 shown above). Thanks for catching this :)

Reply:
I agree that it could get confusing for developers using these components, especially with typescript, since these props will show up in editors, but not all props are supported.

idk if this is overkill, but we could split these (direction/type) into their own types so that icons that support these props can define their props as `IconProps & Direction & Type`

I think we should optimize for the consumers of these components rather than the creators.